src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex.cs

```diff
@@ -16,29 +16,29 @@ internal abstract partial class AbstractSyntaxIndex<TIndex>
         where TIndex : AbstractSyntaxIndex<TIndex>
     {
         protected delegate TIndex? IndexReader(StringTable stringTable, ObjectReader reader, Checksum? checksum);
-        protected delegate TIndex IndexCreator(Document document, SyntaxNode root, Checksum checksum, CancellationToken cancellationToken);
+        protected delegate TIndex IndexCreator(Project project, SyntaxNode root, Checksum checksum, CancellationToken cancellationToken);
 
-        private static readonly ConditionalWeakTable<Document, TIndex?> s_documentToIndex = new();
-        private static readonly ConditionalWeakTable<DocumentId, TIndex?> s_documentIdToIndex = new();
+        private static readonly ConditionalWeakTable<DocumentState, TIndex?> s_documentStateToIndex = new();
```

Comment:
Same here.  We now associate these *syntax* index with a DocumentState, not Document instance.  No reason to dump them when an edit happens to an unrelated document elsewhere.